### ISSUE TITLE
New Multiplayer Levels require going to ladder after submitting

### ISSUE THREAD

**Issue by nemoyatpeace** [codecombat/codecombat#1741]
When playing one of the new levels and using the submit button, after it runs I'm met with this dialog:
![image](https://cloud.githubusercontent.com/assets/6550021/4992370/722473f8-69da-11e4-8e06-ffdf7afc70b4.png)

I don't have the option to just return to the code, I'm forced to return to the ladder, then reload the page.

**Comment by sotonin** [comment]
Yep. seen this as well. I just press escape though

**Comment by UltCombo** [comment]
Even after so much redesigning, this is still a (frustrating) issue, by the way. The `Esc` workaround works, but it is non-intuitive.

Also, there's no way to get back to the world map, the multiplayer game view only has a button to go back to the ladder view.

**Comment by nwinter** [comment]
You're right. We have another redesign coming for this focusing on the user experience; we've got some wireframes together, just need to find time to build it.

**Comment by UltCombo** [comment]
Oh, nice to hear that. :smile:
I've just opened an unrelated user experience issue (#3215), I wonder if the next redesign already covers that?

**Comment by Harry-the-Wanderer** [comment]
After pressing `Esc` you can then press the browsers back button and it should load up the campaign you were just in before starting the level.  

Now I don't know if causes issues with how the game state is being tracked by the web application, but it should at least get the student back to the campaign area.

**Comment by UltCombo** [comment]
@Harry-the-Wanderer CoCo is not a SPA and does not intercept the back button navigation as far as I know. When navigating back to the campaign view, I believe the application sends an uncached request to fetch the player information, so it should work properly.

**Comment by Harry-the-Wanderer** [comment]
This might be a hack, but why not just create a hyperlinked button on the multiplayer level complete screen that just creates a browsers back button effect.  So when you click on the "button" all it really does is a window.history.back(); effect or similar.

**Comment by UltCombo** [comment]
@Harry-the-Wanderer the problem with `history.back()` is that it won't work when you open the level in a new tab. I guess it could work for most cases, though it is not the ideal solution.

**Comment by nwinter** [comment]
Actually, CoCo *is* a single-page web application (using Backbone) and does not perform a new page load when you go back, instead keeping a lot of data that it had loaded previously.

**Comment by UltCombo** [comment]
Oh, that's good to know, my bad! 😅

I knew about Backbone, but really, CoCo does not feel very much like a SPA—the UI doesn't shake around nor slows down when transitioning views, there's hardly any of the bugs or quirks typically found in SPAs (such as memory leaks, broken back/forward navigation or accumulated event listeners). I may have to take a second look at Backbone sometime soon.
